napari/_qt/widgets/qt_viewer_status_bar.py

```diff
@@ -15,7 +22,49 @@ class ViewerStatusBar(QStatusBar):
     def __init__(self, parent: '_QtMainWindow') -> None:
         super().__init__(parent=parent)
 
-        self.showMessage(trans._('Ready'))
+        main_widget = QWidget()
+
+        layout = QHBoxLayout()
+
+        self._status = QLabel('Ready')
+        self._status.setMinimumSize(0, 16)
+        self._status.setSizePolicy(QSizePolicy.Minimum, QSizePolicy.Maximum)
+        self._layer_base = QElidingLabel(trans._(''))
+        self._layer_base.setElideMode(Qt.TextElideMode.ElideMiddle)
+        self._layer_base.setMinimumSize(120, 16)
+        self._layer_base.setSizePolicy(
+            QSizePolicy.Minimum, QSizePolicy.Maximum
+        )
+
+        self._plugin_reader = QElidingLabel(trans._(''))
+        self._plugin_reader.setMinimumSize(80, 16)
+        self._plugin_reader.setSizePolicy(
+            QSizePolicy.Minimum, QSizePolicy.Maximum
+        )
+
+        self._plugin_reader.setElideMode(Qt.TextElideMode.ElideMiddle)
+        self._source_type = QLabel('')
+        self._source_type.setMinimumSize(50, 16)
+        self._source_type.setSizePolicy(
+            QSizePolicy.Minimum, QSizePolicy.Maximum
+        )
+
+        self._coordinates = QLabel('')
+        self._coordinates.setMinimumSize(80, 16)
+        self._coordinates.setSizePolicy(
+            QSizePolicy.Minimum, QSizePolicy.Maximum
+        )
+
+        layout.addWidget(self._status)
+        layout.addWidget(self._layer_base)
+        layout.addWidget(self._source_type)
+        layout.addWidget(self._plugin_reader)
+        layout.addWidget(self._coordinates)
+        layout.addStretch(0)
```

Comment:
I would also use `layout.setSpacing(0)` here to squish up the space in between each layout widget.